### PR TITLE
Implement actions for selected objects #146

### LINKED ISSUE #146: Implement actions for selected objects
As a player having selected tree I want to be able to collect the wood from it.
Analogically for copper etc.
Lumberjacking (collecting wood)
Mining (collecting copper)

### PR DESCRIPTION
**Pull request by neu5**

### CODE REVIEW

File: client/events/sockets.js [hunk 1]
```diff
@@ -160,6 +172,19 @@ export default (game) => {
...
+  game.socket.on("player:action", ({ name }) => {
+    if (name) {
+      game.profile.setActionButton(name);
+    } else {
+      game.profile.resetActionButton();
+    }
+  });
+
+  game.socket.on("player:backpack", (value) => {
+    console.log("backpack");
```
Comment (przenz) [inline client/events/sockets.js]: Remove

File: client/index.html [hunk 2]
```diff
@@ -34,6 +34,7 @@
                         <legend>Selected object</legend>
                         <label class="selected__name"></label>
                         <button class="selected__drop-button" disabled>Drop</button>
```
Comment (przenz) [inline client/index.html]: This is kind of misleading now, should probably change button text to "unselect"

File: client/ui/profile.js [hunk 3]
```diff
@@ -1,20 +1,26 @@
-const WEAPON_PATH = "/assets/gfx/";
+const GFX_PATH = "/assets/gfx/";
 const WEAPONS = {
   bow: "bow.png",
   sword: "sword.png",
 };
+const ITEMS = {
+  wood: "wood.png",
+  "copper ore": "copper-ore.png",
```
Comment (przenz) [inline client/ui/profile.js:8]: Is this space here safe? Maybe `copper_ore` instead?

File: server/gameObjects/Player.js [hunk 4]
```diff
@@ -94,6 +102,27 @@ class Player {
     this.toRespawn = false;
   }
 
+  addToBackpack(newItem, action) {
+    const item = this.backpack.items.find((i) => i.name === newItem);
+
+    if (item) {
+      item.quantity += 1;
+      this.energyUse(action);
```
Comment (przenz) [inline server/gameObjects/Player.js]: Not sure if this place is best for energy burn as not all actions will be backpack related, move to socket emit?

File: server/index.js [hunk 5]
```diff
@@ -140,6 +152,29 @@ io.on("connection", (socket) => {
...
+      }
+    });
+
+    socket.on("action:start", ({ name }) => {
+      const player = players.get(name);
+
+      const item = getItem(player.selectedPlayer);
+
+      if (!player.canPerformAction()) {
+        // send message that action can't be performed
```
Comment (przenz) [inline server/index.js]: Let's add TODO with new issue #id

**Review by przenz — CHANGES_REQUESTED** [review]

File: client/ui/profile.js (hunk 3 above)
Comment (neu5) [reply client/ui/profile.js:8]: It is safe
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Property_accessors

**Review by przenz — APPROVED** [review]

**Pull request merged**